tests/src/Kernel/Entity/GroupMembershipManagerTest.php

```diff
@@ -318,4 +329,315 @@ public function groupContentProvider() {
+            'expected_memberships' => [2],
+          ],
+          // There are no pending or blocked moderators.
+          [
+            'roles' => ['moderator'],
+            'states' => [
+              OgMembershipInterface::STATE_BLOCKED,
+              OgMembershipInterface::STATE_PENDING,
+            ],
+            'expected_memberships' => [],
+          ],
+        ],
+      ],
+      'entity_test' => [
+        0 => [
+          // The first test entity group doesn't have any moderators or admins.
+          // Check that duplicated array values doesn't affect the result.
+          [
+            'roles' => [
+              'moderator',
+              OgRoleInterface::ADMINISTRATOR,
+              'moderator',
+              'moderator',
+              OgRoleInterface::ADMINISTRATOR,
+            ],
+            'states' => [
+              OgMembershipInterface::STATE_ACTIVE,
+              OgMembershipInterface::STATE_BLOCKED,
+              OgMembershipInterface::STATE_PENDING,
+              OgMembershipInterface::STATE_PENDING,
+              OgMembershipInterface::STATE_BLOCKED,
+              OgMembershipInterface::STATE_ACTIVE,
+            ],
+            'expected_memberships' => [],
+          ],
+        ],
+        // Check active members.
+        [
+          'roles' => [
+            OgRoleInterface::AUTHENTICATED,
+          ],
+          'states' => [
+            OgMembershipInterface::STATE_ACTIVE,
+          ],
+          'expected_memberships' => [0, 3],
+        ],
+        1 => [
+          // There are two blocked users in the second test entity group.
+          [
+            'roles' => [
+              OgRoleInterface::AUTHENTICATED,
+              OgRoleInterface::ADMINISTRATOR,
+              'moderator',
+            ],
+            'states' => [
+              OgMembershipInterface::STATE_BLOCKED,
+            ],
+            'expected_memberships' => [4, 7],
+          ],
+          // There is one blocked moderator.
+          [
+            'roles' => [
+              OgRoleInterface::ADMINISTRATOR,
+              'moderator',
+            ],
+            'states' => [
+              OgMembershipInterface::STATE_BLOCKED,
+            ],
+            'expected_memberships' => [4],
+          ],
+        ],
+      ],
+    ];
+
+    foreach ($matrix as $entity_type_id => $groups) {
+      foreach ($groups as $group_key => $test_cases) {
+        foreach ($test_cases as $test_case) {
+          $group = $this->groups[$entity_type_id][$group_key];
+          $role_names = $test_case['roles'];
+          $states = $test_case['states'];
+          $expected_memberships = $test_case['expected_memberships'];
+
+          /** @var \Drupal\og\OgMembershipInterface[] $actual_memberships */
+          $actual_memberships = $this->membershipManager->getGroupMembershipsByRoleNames($group, $role_names, $states);
+
+          $this->assertEquals(count($expected_memberships), count($actual_memberships));
```

Comment:
@brummbar, you can use the github suggestion widget to do suggestions like this.
```suggestion
          $this->assertCount(count($expected_memberships), $actual_memberships);
```

Reply:
Cool, didn't know this functionality.

Reply:
@MPParsley I didn't know this feature either, it looks very handy! Unfortunately the suggestion wasn't correct :-/

I went with the `$this->assertSameSize()`, that is the perfect match for our use case!

Reply:
This new suggestion feature seems to have some bugs regarding the CI pipelines, the build was started but immediately canceled. It's still in beta, I'm sure they'll iron this out soon enough.

Reply:
@pfrenssen indeed, assertSameSize() is better.

Thanks for the update about the CI pipeline. I'm not sure how it's set up on this project but on our projects Travis is triggered (and runs) as if it were a regular commit.

This feature is going to be a great help for open source collaboration as the threshold is much lower to contribute (especially when compared to an archaic d.o patch workflow...).